src/integrations/prefect-docker/prefect_docker/deployments/steps.py

```diff
@@ -74,16 +80,44 @@ class PushDockerImageResult(TypedDict):
     """
 
     image_name: str
-    tag: str
+    tag: Optional[str]
     image: str
-    additional_tags: Optional[str]
+    additional_tags: Optional[List[str]]
+
+
+def _make_hashable(obj):
+    if isinstance(obj, dict):
+        return json.dumps(obj, sort_keys=True)
+    elif isinstance(obj, list):
+        return tuple(_make_hashable(v) for v in obj)
+    return obj
 
 
+def cacheable(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        ignore_cache = kwargs.pop("ignore_cache", False)
+        key = (
+            tuple(_make_hashable(arg) for arg in args),
+            tuple((k, _make_hashable(v)) for k, v in sorted(kwargs.items())),
+        )
+        if ignore_cache or key not in STEP_OUTPUT_CACHE:
+            logger.debug(f"Cache miss for {func.__name__}, running function.")
+            STEP_OUTPUT_CACHE[key] = func(*args, **kwargs)
+        else:
+            logger.info(f"Cache hit for {func.__name__}, returning cached value.")
```

Comment:
The way this log message is written makes me think it would work better as a debug level log.
```suggestion
            logger.debug(f"Cache hit for {func.__name__}, returning cached value.")
```